src/Api/BrandsApi.php

```diff
@@ -0,0 +1,350 @@
+     * @param string        $email;
+     * @param string|null   $url;
+     * @param string        $voice;
+     * @param string|null   $fax;
+     * @param string|null   $privacyContact;
+     * @param string        $abuseContact;
+     * @param Carbon        $createdDate;
+     * @param Carbon|null   $updatedDate;
+     */
+    public function update(
+        string $customer,
+        string $handle,
+        ?string $locale,
+        ?string $organization,
+        ?array $addressLine,
+        ?string $postalCode,
+        ?string $city,
+        ?string $state,
+        ?string $country,
+        ?string $email,
+        ?string $url,
+        ?string $voice,
+        ?string $fax,
+        ?string $privacyContact,
+        ?string $abuseContact,
+        ?Carbon $createdDate,
+        ?Carbon $updatedDate
+    ): void {
+        $payload = [];
+
+        if ($locale) {
+            $payload['locale'] = $locale;
+        }
+        if ($organization) {
+            $payload['organization'] = $organization;
+        }
+        if ($addressLine) {
+            $payload['addressLine'] = $addressLine;
+        }
+        if ($postalCode) {
+            $payload['postalCode'] = $postalCode;
+        }
+        if ($city) {
+            $payload['city'] = $city;
+        }
+        if ($state) {
+            $payload['state'] = $state;
+        }
+        if ($country) {
+            $payload['country'] = $country;
+        }
+        if ($email) {
+            $payload['email'] = $email;
+        }
+        if ($url) {
+            $payload['url'] = $url;
+        }
+        if ($voice) {
+            $payload['voice'] = $voice;
+        }
+        if ($fax) {
+            $payload['fax'] = $fax;
+        }
+        if ($privacyContact) {
+            $payload['privacyContact'] = $privacyContact;
+        }
+        if ($abuseContact) {
+            $payload['abuseContact'] = $abuseContact;
+        }
+        if ($createdDate) {
+            $payload['createdDate'] = $createdDate;
+        }
+        if ($updatedDate) {
+            $payload['updatedDate'] = $updatedDate;
+        }
+
+        $this->client->post("/v2/customers/{$customer}/brands/{$handle}/update", $payload);
+    }
+
+    /**
+     * @see https://dm.realtimeregister.com/docs/api/brands/delete
+     *
+     * @param string $customer
+     * @param string $handle
+     */
+    public function delete(string $customer, string $handle): void
+    {
+        $this->client->delete("/v2/customers/{$customer}/brands/{$handle}");
+    }
+
+    /**
+     * @see https://dm.realtimeregister.com/docs/api/brands/templates/get
+     *
+     * @param string $customer
+     * @param string $brand
+     * @param string $name
+     *
+     * @return Template
+     */
+    public function getTemplate(string $customer, string $brand, string $name): Template
+    {
+        return Template::fromArray($this->client->get("/v2/customers/{$customer}/brands/{$brand}/templates/{$name}")->json());
```

Comment:
It would be nice if you would validate the `$name` against the template name enum here.

Reply:
Added: https://github.com/sandwave-io/realtimeregister-php/pull/25/commits/a1f9ef0d455c9ebad2b5cf1b762252a8ed3b12ea